scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
Since I am not a Fortran expert, I can not say anything about that.

In the Matrix Market format description http://math.nist.gov/MatrixMarket/reports/MMformat.ps from http://math.nist.gov/MatrixMarket/formats.html, it is only specified that the data has to be written as floats. But it is not specified that this has to be in fixed-point format or exponential format. Therefore, I see no reason why it must be the exponential format even if it was used in the referenced implementations.

As I said, I think the fixed-point format is more intuitive in the context of  precision. By changing this, no incompatibility would arise because the cast back to float in the read method would work with both formats.